posts/2020-11-26-MP-rest-client-Kubernetes-secrets-21001.adoc

```diff
@@ -0,0 +1,216 @@
+* <<kube, Using Kubernetes secrets in Liberty variables>>
+
+
+[#MP]
+=== MP Rest Client 2.0
+
+MicroProfile REST Client is a type-safe client API enabling rapid development of applications capable of consuming RESTful services. Version 2.0 is the latest update and add support for HTTP proxy servers, automatically following HTTP redirects, additional configuration options for JSON-B providers and how multiple query parameters should be specified, and support for Server Sent Events.
+
+To enable this feature, add `<feature>mpRestClient-2.0</feature>` to you list of features in the `<featureManager>` element - the `microprofile-4.0` convenience feature will also enable this feature.
+
+For examples of how to these new features, please refer to the link:http://download.eclipse.org/microprofile/microprofile-rest-client-2.0-RC2/microprofile-rest-client-2.0-RC2.html[specification document] or the link:https://github.com/eclipse/microprofile-rest-client[MP Rest Client project page].
+
+
+[#kube]
+=== Using Kubernetes secrets in Liberty variables
+
+In a cloud environment, sensitive information such as passwords and OAuth tokens may be stored in Kubernetes secrets. To access these secrets today in Liberty via configuration variables they would need to be exposed as environment variables in a Kubernetes pod. This update will allow configuration variables to be automatically populated from Kubernetes secrets without having to expose them as environment variables.
+
+To make use of this function, the Kubernetes secrets need to be mapped to the file system, either by using the service binding operator or by mounting the secrets to a volume in the pod definition. Files will be read from the location defined in the `SERVICE_BINDING_ROOT` environment variable, which defaults to `WLP_CONFIG_DIR`/bindings. The variable name will be taken from the name of the file, and the value will be the contents of the file.
+
+For example, we may have a Kubernetes secret named 'account_database' that contains a username and password:
+
+[.img_border_dark]
+image::img/blog/blog_21001_beta_username_password.png[align="center",Image of Kubernetes secret named 'account_database']
+
+The service binding operator can automatically map that secret to the file system, or it could be done manually in a deployment definition:
+
+[.img_border_dark]
+image::img/blog/blog_21001_deployment_definition.png[align="center",Image of Kubernetes secret named 'account_database']
+
+The above definition will result in the files /bindings/account_db/username and /bindings/account_db/password being created when the pod is created. Liberty will create variables from both files, and users can access them using normal Liberty variable syntax, `${account_db/username}` and `${account_db/password}`.
+
+For more Information:
+* link:https://github.com/k8s-service-bindings/spec[Service Binding specification]
+* link:https://kubernetes.io/docs/concepts/configuration/secret/[Kubernetes Secrets]
```

Comment:
not rendering correctly; each link should be on a new line (indent or bullet)

Reply:
Done